Update PR Template: Refactor Checklist (#12063)
dependabot/dependabot-core: .github/pull_request_template.md

- [ ] I have run the complete test suite to ensure all tests and linters pass.
- [ ] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.

> - [ ] I have run Sorbet type-checking in strict mode to ensure that all types are correct and complete.

Reviewer: I think we should enforce this via CI check, and not via a human.

It's yet another hurdle for a drive-by contributor... most will have no idea how to run sorbet checks, and if their PR is fine, then _not_ mentioning it here avoids confusion.

In CI, if it fails, then part of the error message should say "run X command to fix this"

Author: @jeffwidman , 

Is it possible to have 2 templates one for contributors and one for maintainers?

Author: Is it possible to set up separate templates for contributors and maintainers, and for maintainers, make Sorbet typings essential?

For example:

```
.github/PULL_REQUEST_TEMPLATE/contributor_template.md
.github/PULL_REQUEST_TEMPLATE/maintainer_template.md
```

Author: For contributors: 

```
- [ ] I have run tests, verified functionality, and added tests for new features.
- [ ] I have written clear commit messages and provided a detailed PR description, including any breaking changes.
- [ ] I have ensured the code is well-documented and follows project standards.
```

For maintainers:

```
- [ ] I have run tests, verified functionality, and added tests for new features.
- [ ] I have written clear commit messages and provided a detailed PR description, including any breaking changes.
- [ ] I have ensured the code is well-documented, follows project standards, and enforces Sorbet typing where necessary.
```

- [ ] I have written clear and descriptive commit messages.
- [ ] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [ ] I have ensured that the code is well-documented and easy to understand.

> - [ ] I have ensured that there are no breaking changes unless explicitly stated and handled.
> - [ ] I have checked the backward compatibility of my changes.

Reviewer: I think the intent of this is great, but I don't think this is a realistic expectation for our drive-by open source contributors.

IMO this should at least partially be the responsibility of the maintainers to eyeball for compatibility.

Instead, I'd word this as a reminder that "if you're aware of 🦶 🔫 , please call it out"

```suggestion
- [ ] I have called out any breaking changes that I'm aware of.
```
